docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -0,0 +1,420 @@
+    {
+        new SecretProperties("secret5"),
+        new SecretProperties("secret6")
+    },
+    continuationToken: null,
+    Mock.Of<Response>());
+
+Pageable pageable = Pageable.FromPages(new[] { page1, page2, lastPage });
+AsyncPageable asyncPageable = AsyncPageable.FromPages(new[] { page1, page2, lastPage });
+```
+
+## Write a mocked unit test
+
+Suppose your app contains a class that finds the names of keys that will expire within a given amount of time.
+
+```csharp
+using Azure.Security.KeyVault.Secrets;
+
+public class AboutToExpireSecretFinder
+{
+    private readonly TimeSpan _threshold;
+    private readonly SecretClient _client;
+
+    public AboutToExpireSecretFinder(TimeSpan threshold, SecretClient client)
+    {
+        _threshold = threshold;
+        _client = client;
+    }
+
+    public async Task<string[]> GetAboutToExpireSecrets()
+    {
+        List<string> secretsAboutToExpire = new();
+
+        await foreach (var secret in _client.GetPropertiesOfSecretsAsync())
+        {
+            if (secret.ExpiresOn.HasValue &&
+                secret.ExpiresOn.Value - DateTimeOffset.Now <= _threshold)
+            {
+                secretsAboutToExpire.Add(secret.Name);
+            }
+        }
+
+        return secretsAboutToExpire.ToArray();
+    }
+}
+```
+
+You want to test the following behaviors of the `AboutToExpireSecretFinder` to ensure they continue working as expected:
+
+* Secrets that don't have an expiry date set are not returned.
+* Secrets with an expiry date closer to the current date than the threshold are returned.
+
+When unit testing you only want the unit tests to verify the application logic and not whether the Azure service or SDK works correctly. The following example tests the key behaviors using the popular xUnit framework for C#:
+
+```csharp
+using Azure;
+using Azure.Security.KeyVault.Secrets;
+using Moq;
+using Xunit;
+
+public class AboutToExpireSecretFinderTests
+{
+    [Fact]
+    public async Task DoesNotReturnNonExpiringSecrets()
+    {
+        // Arrange
+        // Create a page of enumeration results
+        Page<SecretProperties> page = Page<SecretProperties>.FromValues(new[]
+        {
+            new SecretProperties("secret1") { ExpiresOn = null },
+            new SecretProperties("secret2") { ExpiresOn = null }
+        }, null, Mock.Of<Response>());
+
+        // Create a pageable that consists of a single page
+        AsyncPageable<SecretProperties> pageable = AsyncPageable<SecretProperties>.FromPages(new[] { page });
+
+        // Setup a client mock object to return the pageable when GetPropertiesOfSecretsAsync is called
+        var clientMock = new Mock<SecretClient>();
+        clientMock.Setup(c => c.GetPropertiesOfSecretsAsync(It.IsAny<CancellationToken>()))
+            .Returns(pageable);
+
+        // Create an instance of a class to test passing in the mock client
+        var finder = new AboutToExpireSecretFinder(TimeSpan.FromDays(2), clientMock.Object);
```

Comment:
```suggestion
        AboutToExpireSecretFinder finder = new(TimeSpan.FromDays(2), clientMock.Object);
```

Reply:
1. I think you shouldn't use `var`. See comment above.
2. I don't think you should use `new(...)`. See same comment above.

This is a fairly basic concept and I think it behooves readers to be as plain as possible - to avoid a lot of newer C#-isms.